packages/legend-studio-extension-query-builder/src/components/QueryBuilder_LegendStudioPlugin.tsx

```diff
@@ -222,10 +222,11 @@ export class QueryBuilder_LegendStudioPlugin
                 }),
               );
               if (queryBuilderExtension.mode) {
+                queryBuilderExtension.queryBuilderState.resetApp();
```

Comment:
see my other comment about order of tree data refresh vs. setting class.

Reply:
Logic will be moved to `changeClass` see overall comment